### DISCUSSION TITLE
Startup with wrong configuration

### DISCUSSION THREAD

**Discussion by Rillke** [nginx-proxy/nginx-proxy#1847] [Q&A] [answered]
Hi,

today, after a VM reboot, this nginx-proxy did startup with an inappropriate configuration leading to requests being directed to the wrong container. After manually running `docker-compose restart nginx-proxy` the correct configuration has been generated.

Here is such a request (log):

```
nginx-proxy_1  | nginx.1     | 2021/12/20 06:37:01 [error] 46#46: *21639 connect() failed (111: Connection refused) while connecting to upstream, client: 1.1.1.1, server: a.example.com, request: "GET /search/episode.xml?limit=6&offset=0 HTTP/1.1", upstream: "http://172.19.0.6:8080/search/episode.xml?limit=6&offset=0", host: "a.example.com"
```

`docker network inspect`:

```json
            "3dc64bbc858baa0412cb68xxx": {
                "Name": "software-docker_container1_1",
                "EndpointID": "016d71c705xxx",
                "MacAddress": "02:xx:xx:13:00:06",
                "IPv4Address": "172.19.0.6/16",
                "IPv6Address": ""
            },
            "a0ee561716141553677c73xxx": {
                "Name": "software-docker_container2_1",
                "EndpointID": "cea7d59828675xxx",
                "MacAddress": "02:xx:xx:13:00:02",
                "IPv4Address": "172.19.0.2/16",
                "IPv6Address": ""
            },
            "bed9bdc7e0398fdxxx": {
                "Name": "software-docker_nginx-proxy_1",
                "EndpointID": "c5bb23758afxxx",
                "MacAddress": "02:xx:xx:13:00:03",
                "IPv4Address": "172.19.0.3/16",
                "IPv6Address": ""
            },
```

While the request for `a.example.com` was supposed to go to `172.19.0.2:8080`, it was sent to `172.19.0.6:8080` (this is what I get from the above request log's `upstream` field).

Here are the relevant variables for each container:

`software-docker_container1_1`:

```
VIRTUAL_PORT=9000
VIRTUAL_PROTO=fastcgi
VIRTUAL_HOST=3.3.3.3
```

(container1's VIRTUAL_HOST equals the host's public IP address)

`software-docker_container2_1`:

```
VIRTUAL_PORT=8080
VIRTUAL_HOST=a.example.com
```
`software-docker_nginx-proxy_1`:

```
DEFAULT_HOST=a.example.com
```

That's the first time I'm experiencing this issue in production. Have there been any regressions recently or am I using it wrong?

Thanks!

**Comment by buchdag** [comment, marked as answer]
Hi

> (container1's VIRTUAL_HOST equals the host's public IP address)

I don't recommend doing this. Even if it does work, it's never been officially supported nor tested, and is not the intended use of the container.

That said I've experienced issues similar to yours (a request being proxied to an correct container, with the misconfiguration being after a container restart) in the past without using IP in place of proper host name in `VIRTUAL_HOST`. I've never been able to reliably reproduce the issue and it's always been a very uncommon occurrence.

To my knowledge there hasn't been any recent known regression that could have caused this.
